ui/oopsyraidsy/data/06-ew/raid/p10n.ts

```diff
@@ -1,12 +1,118 @@
+import NetRegexes from '../../../../../resources/netregexes';
 import ZoneId from '../../../../../resources/zone_id';
 import { OopsyData } from '../../../../../types/data';
 import { OopsyTriggerSet } from '../../../../../types/oopsy';
 
-export type Data = OopsyData;
+// TODO: Replace Unmitigated Explosion by detecting/blaming whomever did not soak a pillar.
+
+export interface Data extends OopsyData {
+  bindingSoulPlayers?: string[];
+  concentratedPoisonPlayers?: string[];
+}
 
 const triggerSet: OopsyTriggerSet<Data> = {
   zoneId: ZoneId.AnabaseiosTheTenthCircle,
-  damageWarn: {},
+  damageWarn: {
+    'P10N Parted Plumes Blade': '8271', // rotating blade AoE
+    'P10N Pandaemoniac Ray': '8266', // half-room AoE
+  },
+  damageFail: {
+    'P10N Unmitigated Explosion': '8260', // unsoaked pillar
+    'P10N Imprisonment': '8262', // circle AoE from jails
+    'P10N Cannonspawn': '8264', // donut AoE from jails
+  },
+  gainsEffectWarn: {
+    'P10N Parted Plumes Puddle': 'C05', // Bleeding debuff from standing in center puddle
+  },
+  soloWarn: {
+    'P10N Pandaemoniac Meltdown': '8277', // line stack
+  },
+  triggers: [
+    {
+      // Stacking debuff that auto-increments and upon reaching 5 stacks is fatal.
+      id: 'P10N Binding Soul Share Gain',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'DDB' }),
+      mistake: (data, matches) => {
+        data.bindingSoulPlayers ??= [];
+        if (!data.bindingSoulPlayers.includes(matches.target)) {
+          data.bindingSoulPlayers.push(matches.target);
+          return {
+            type: 'fail',
+            blame: matches.target,
+            reportId: matches.targetId,
+            text: matches.effect,
+          };
+        }
+      },
+    },
+    {
+      id: 'P10N Binding Soul Share Loss',
+      type: 'LosesEffect',
+      netRegex: NetRegexes.losesEffect({ effectId: 'DDB' }),
+      run: (data, matches) => {
+        data.bindingSoulPlayers = (data.bindingSoulPlayers ??= []).filter((p) =>
+          p !== matches.target
+        );
+      },
+    },
+    {
+      // Stacking debuff that auto-increments as you remain in the poison.
+      id: 'P10N Concentrated Poison Gain',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'E6C' }),
```

Comment:
Ahh, I was not aware of the `count` property. I think `gainsEffectWarn` works for Concentrated Poison, since stacks increase as you continue to stand in the poison (and should be continually reported as a mistake), but I think using `count` for Binding Soul Share makes sense since the debuff auto-increments and it gets spammy.